src/workerd/jsg/jsg.h

```diff
@@ -1984,6 +1984,34 @@ class Lock {
   // Returns a random UUID for this isolate instance. This is largely intended for logging and
   // diagnostic purposes.
 
+  enum class HandleScopeOptions {
+    NONE,
+    ENTER_CONTEXT,
+  };
+
+  auto withinHandleScope(auto fn, HandleScopeOptions options = HandleScopeOptions::NONE) {
```

Comment:
If you use `auto&& fn`, it allows the caller to pass an lvaule reference to a long-lived functor if they want, e.g. a lambda separately declared on the stack, without having to `kj::mv()` it.

Admittedly I don't know why anyone would want to do that with this function, but my usual pattern when writing functions that take lambda parameters is to allow that.